connector.go

```diff
@@ -81,6 +81,12 @@ func (c *SQLConnection) GetDatabase() *sqlx.DB {
 	}
 
 	var err error
+
+	userinfo := c.URL.User.Username()
+	if password, hasPassword := c.URL.User.Password(); hasPassword {
+		userinfo = userinfo + ":" + password
```

Comment:
Does this work when userinfo or password have a `:`?

Reply:
This works when password has a `:`, but fails when username does.  It looks like this is related to https://github.com/go-sql-driver/mysql/issues/591 - their DSN parser doesn't support it.

Reply:
to be clear, the existing implementation also fails when the username has a :, and this does not fix that.  this does fix when the username or password has certain other characters

Reply:
Ah, didn't know that, thanks!